File: LeetCode/Medium/2924-find-champion-ii/2924-find-champion-ii.java
class Solution {
    public int findChampion(int n, int[][] edges) {
        int[] degree = new int[n];

        for (int[] edge : edges) {
            degree[edge[1]]++;
        }

        int champ = -1;
        int champCount = 0;

        for (int i = 0; i < n; i++) {
            if (degree[i] == 0) {
                champCount++;
                champ = i;
            }
        }

        return champCount > 1 ? -1 : champ;
    }
}

File: LeetCode/Medium/2924-find-champion-ii/2924-find-champion-ii.kt
class Solution {
    fun findChampion(
        n: Int,
        edges: Array<IntArray>,
    ): Int {
        val degree = IntArray(n)

        for (edge in edges) {
            degree[edge[1]]++
        }

        var champ = -1
        var champCount = 0

        for (i in 0 until n) {
            if (degree[i] == 0) {
                champCount++
                champ = i
            }
        }

        return if (champCount > 1) -1 else champ
    }
}
